lang/cs/Org.Apache.REEF.Network/Elastic/Failures/Default/DefaultFailureStateMachine.cs

```diff
@@ -0,0 +1,309 @@
+
+        private readonly SortedDictionary<DefaultFailureStates, DefaultFailureStates> transitionMapUp = new SortedDictionary<DefaultFailureStates, DefaultFailureStates>()
+        {
+            { DefaultFailureStates.Continue, DefaultFailureStates.ContinueAndReconfigure },
+            { DefaultFailureStates.ContinueAndReconfigure, DefaultFailureStates.ContinueAndReschedule },
+            { DefaultFailureStates.ContinueAndReschedule, DefaultFailureStates.StopAndReschedule },
+            { DefaultFailureStates.StopAndReschedule, DefaultFailureStates.Fail }
+        };
+
+        private readonly SortedDictionary<DefaultFailureStates, DefaultFailureStates> transitionMapDown = new SortedDictionary<DefaultFailureStates, DefaultFailureStates>()
+        {
+            { DefaultFailureStates.ContinueAndReconfigure, DefaultFailureStates.Continue },
+            { DefaultFailureStates.ContinueAndReschedule, DefaultFailureStates.ContinueAndReconfigure },
+            { DefaultFailureStates.StopAndReschedule, DefaultFailureStates.ContinueAndReschedule },
+            { DefaultFailureStates.Fail, DefaultFailureStates.StopAndReschedule }
+        };
+
+        private readonly IDictionary<DefaultFailureStates, float> transitionWeights = new Dictionary<DefaultFailureStates, float>()
+        {
+            { DefaultFailureStates.ContinueAndReconfigure, 0.01F },
+            { DefaultFailureStates.ContinueAndReschedule, 0.40F },
+            { DefaultFailureStates.StopAndReschedule, 0.60F },
+            { DefaultFailureStates.Fail, 0.80F }
+        };
+
+        private static List<int> canMoveToComplete = new List<int>()
+        {
+            (int)DefaultFailureStates.Continue,
+            (int)DefaultFailureStates.ContinueAndReconfigure,
+            (int)DefaultFailureStates.ContinueAndReschedule,
+            (int)DefaultFailureStates.Complete
+        };
+
+        private static List<int> isFinalState = new List<int>()
+        {
+            (int)DefaultFailureStates.Complete
+        };
+
+        /// <summary>
+        /// Default failure state machine starting with 0 data points and in continue state.
+        /// </summary>
+        [Inject]
+        public DefaultFailureStateMachine() : this(0, DefaultFailureStates.Continue)
+        {
+        }
+
+        /// <summary>
+        /// Default failure stata machine starting with a given amount of data points and a given intial state.
+        /// </summary>
+        /// <param name="initalPoints">The number of initial data points for the machine, 0 by default</param>
+        /// <param name="initalState">The initial state, continue by default</param>
+        public DefaultFailureStateMachine(int initalPoints = 0, DefaultFailureStates initalState = DefaultFailureStates.Continue)
+        {
+            NumOfDataPoints = initalPoints;
+            NumOfFailedDataPoints = initalPoints;
+            State = new DefaultFailureState((int)initalState);
```

Comment:
we can define constructor
```C#
public DefaultFailureState(DefaultFailureStates state)
```
to avoid the explicit cast here

Reply:
Mmm not sure. Because I will have to do a cast later on because FailureState is an int (cannot be a DefaultFailureState because it's defined in the base class).